adr/0005-use-protobuf-to-define-and-generate-contracts.md

```diff
@@ -0,0 +1,23 @@
+# 5. Use Protocol Buffers to define and generate contracts
+
+Date: 2022-11-17
+
+## Status
+
+Accepted
+
+## Context
+
+We need to define contracts between the components of Trento to communicate with each other.
+We want to use a well-known and widely adopted format to define these contracts.
+We want to use a format that allows us to generate code from the contracts, avoid boilerplate code and reduce the risk of errors.
+
+## Decision
+
+Protocol Buffers is a good candidate for this task, as it is a language-neutral, platform-neutral, extensible mechanism for serializing structured data.
```

Comment:
Could we put a link to protobuf together with the `Protocol Buffers` name?
https://developers.google.com/protocol-buffers/
Or this commonly used `protobuf` term. We are used to this word, and it simplifies questions (like; "Is this protocol buffers "protobuf"")
Yes, I know, we should know that this refers to the `protobuf`

Reply:
protobuf is the short for Protocol Buffers, they are both used. I can do what Wikipedia does: Protocol Buffers (Protobuf) so we all know what we are referring to

Reply:
Yes, I would do that. We use `protobuf` always, so if somebody does some search or anything in the ADRs folder project